components/electric/lib/electric/ddlx/parse/common.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Electric.DDLX.Parse.Common do
+  @doc """
+    uses this to template a compiled regex into each parser with the common bits from below and their own keywords
+  """
+  defmacro __using__(_) do
+    quote do
+      @token_re regex_for_keywords(@keywords)
+      @default_schema "public"
+
+      def token_regex() do
+        @token_re
+      end
+
+      def elements() do
+        @elements
+      end
+    end
+  end
+
+  def scope_and_role(role_name) do
+    if String.contains?(role_name, ":") do
+      String.split(role_name, ":")
+      |> List.to_tuple()
+    else
+      {"__global__", role_name}
+    end
```

Comment:
I'm just nitpicking, but...
```suggestion
    case String.split(role_name, ":") do
      [schema, role] -> {schema, role}
      [role] -> {"__global__", role]
    end
```